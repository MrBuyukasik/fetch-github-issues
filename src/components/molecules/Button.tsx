import React from 'react';
import {
  StyleProp,
  TouchableOpacity as RNTouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

import styled from 'styled-components/native';
import {color} from '../../common/color';
import Text from '../atoms/Text';

const StyledTouchableOpacity = styled(RNTouchableOpacity)`
  height: 50px;
  width: 50%;
  border-radius: 12px;
  background-color: ${color.AQUA_BLUE};
  align-self: center;
`;

const StyledText = styled(Text)`
  text-align: center;
`;

const StyledTextContainer = styled(View)`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

interface Props {
  label: string;
  onClick: () => void;
  style?: StyleProp<ViewStyle> | undefined;
  testID?: string;
}

const Button = (props: Props) => {
  const {style, label, onClick, testID} = props;

  return (
    <StyledTouchableOpacity testID={testID} style={style} onPress={onClick}>
      <StyledTextContainer>
        <StyledText label={label} />
      </StyledTextContainer>
    </StyledTouchableOpacity>
  );
};

export default Button;
