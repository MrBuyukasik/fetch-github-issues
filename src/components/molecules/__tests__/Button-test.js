import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';
fetch = jest.fn(() => Promise.resolve());
const mockFn = jest.fn();

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Button
          label="test"
          testId="button"
          onPress={mockFn}
          accessibilityLabel="button"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
