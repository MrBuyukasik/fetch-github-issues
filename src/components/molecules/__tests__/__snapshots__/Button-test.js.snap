// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#3DBDFF",
      "borderRadius": 12,
      "height": 50,
      "opacity": 1,
      "width": "50%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "width": "100%",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
